src/helics/application_api/MessageFederate.cpp

```diff
@@ -243,6 +256,19 @@ void MessageFederate::registerMessageInterfacesToml(const std::string& tomlStrin
             Endpoint& epObj =
                 (global) ? registerGlobalEndpoint(key, type) : registerEndpoint(key, type);
 
+            loadOptions(this, ept, epObj);
+        }
+    }
+    if (fileops::isMember(doc, "datasink")) {
+        auto& epts = toml::find(doc, "datasink");
+        if (!epts.is_array()) {
+            throw(helics::InvalidParameter("datasink section in toml file must be an array"));
```

Comment:
`datasinks` for consistency with JSON array name, and matching `endpoints` array name in toml.

```suggestion
    if (fileops::isMember(doc, "datasinks")) {
        auto& epts = toml::find(doc, "datasinks");
        if (!epts.is_array()) {
            throw(helics::InvalidParameter("datasinks section in toml file must be an array"));
```